src/mame/sigma/sigmab31.cpp

```diff
@@ -117,9 +119,20 @@ class sigmab31_state : public driver_device
 
 private:
 	void prg_map(address_map &map) ATTR_COLD;
+	void sound_prog_map(address_map &map) ATTR_COLD;
 
 	required_device<cpu_device>     m_maincpu;
 };
+void sigmab52_state::sound_prog_map(address_map &map)
+{
+	map(0x0000, 0x1fff).ram();
+	map(0x6020, 0x6027).rw(m_6840ptm_2, FUNC(ptm6840_device::read), FUNC(ptm6840_device::write));
+//	map(0x6030, 0x6030).w(FUNC(sigmab31_state::audiocpu_irq_ack_w));
+//	map(0x6050, 0x6050).r("soundlatch", FUNC(generic_latch_8_device::read));
+	map(0x6060, 0x6061).rw("ymsnd", FUNC(ym3812_device::read), FUNC(ym3812_device::write));
+	map(0x8000, 0xffff).rom().region("audiocpu", 0);
+}
+
 
 
 void sigmab31_state::prg_map(address_map &map)
```

Comment:
Below the `prg_map` please, that should really be renamed to `main_map` or something ...